tap_google_analytics/client.py

```diff
@@ -232,10 +303,9 @@ def _ensure_access_token(self):
     # Backoff Max Time: try 1 (wait 10) 2 (wait 100) 3 (wait 1000) 4
     # Gives us waits of: (10 * 10 ^ 0), (10 * 10 ^ 1), (10 * 10 ^ 2)
     #
-    # backoff for Timeout exception is already included in "requests.exceptions.RequestException"
-    # as it is the parent class of "Timeout" error
+    # added the backoff for the parent class GoogleAnalyticsClientError and for Timeout error
     @backoff.on_exception(backoff.expo,
-                          (requests.exceptions.RequestException),
+                          (requests.exceptions.Timeout, requests.exceptions.ConnectionError, GoogleAnalyticsClientError),
```

Comment:
Do we need to retry for every error?

Reply:
We are not retrying for every error, we have a giveup function which has specific codes to e retried